### PR TITLE
[WIP] Adds `t2 init --lang rust`

### PR DESCRIPTION
**Pull request by nplus11**
Three things this PR does:

1.	Adds multiple language support to `init`
2.	Adds ability to initialize in a provided directory (aka `t2 init [directory]`)
3.	Cleans up `init` to use promises and path resolution

Todo - could probably still be cleaned up a bit, but good for now

### CODE REVIEW

File: bin/tessel-2.js [hunk 1]
```diff
@@ -11,6 +11,12 @@ var nameOption = {
   help : 'The name of the tessel on which the command will be executed'
 }
 
+var language = {
+  metavar : 'LANG',
+  abbr: 'l',
+  help : 'The language to use <javascript|rust|python>. Javascript by default'
```
Comment (Frijol) [inline bin/tessel-2.js]: Any plans to let these be abbreviated <js | rs | py>?

File: lib/init.js [hunk 2]
```diff
@@ -3,173 +3,275 @@ var Promise = require('bluebird')
...
+    ctx.config = npmConfig;
+
+    // default to auto config
     promzardConfig = path.resolve(__dirname + '/../', 'resources/init-default.js');
     if (options.interactive) {
       promzardConfig = path.resolve(__dirname + '/../', 'resources/init-config.js');
     }
 
-    // Init promozard with appropriate config.
+    // init promozard with appropriate config
```
Comment (Frijol) [inline lib/init.js]: Nitpick: I think the standard is to capitalize

File: lib/init.js [hunk 3]
```diff
@@ -3,173 +3,275 @@ var Promise = require('bluebird')
...
+    var dependencies = [];
+    for (var mod in pkg.dependencies) {
+      dependencies.push(mod + "@" + pkg.dependencies[mod]);
+    }
+    return resolve(dependencies);
+  });
 }
 
 
+// Installs npm
```
Comment (Frijol) [inline lib/init.js]: Doesn't it install the dependencies, not npm itself?

File: lib/init.js [hunk 4]
```diff
@@ -3,173 +3,275 @@ var Promise = require('bluebird')
...
+      });
     }
 
-    // Pipe the contents of the default file into a new file
-    fs.createReadStream(path.resolve(__dirname + './../resources/' + filename))
-      .pipe(fs.createWriteStream(filename));
+  });
+}
 
-    console.log('Wrote \"Hello World\" to index.');
```
Comment (Frijol) [inline lib/init.js]: Is this console.log replaced somewhere? Can't find it.

**Comment by johnnyman727** [comment]
@nplus11 sorry for the slow feedback! Can you go ahead and rebase this?

**Comment by nplus11** [comment]
@johnnyman727 good to go

**Comment by johnnyman727** [comment]
You've got some strange stuff going on here. It looks like all of your commits have been duplicated?

**Comment by nplus11** [comment]
Yeah my local master wasn't up to date when I originally rebased, so updated master and rebased again - this was the result. Not sure why it duped so much, first time I've seen a rebase do that. Regardless it's tested and functional and cleaned up now.

**Comment by johnnyman727** [comment]
@nplus11 it looks like this is still messed up - now it appears there are three copies of commits.

File: bin/tessel-2.js [hunk 5]
```diff
@@ -11,18 +11,58 @@ var nameOption = {
...
+  })
+  .option('lang', language)
+  .option('directory', {
+    position: 1,
+    help: 'Provide a directory to initialize your project'
+  })
+  .help('Initialize repository for your Tessel project')
+
+
+parser.command('list')
```
Comment (johnnyman727) [inline bin/tessel-2.js]: Why does this commit edit how the command line parser accepts the `list` command?

**Comment by johnnyman727** [comment]
Also, it looks like there is some code modified in tessel-2.js that isn't relevant to this commit - can you take those change out? Also, we may want to make the file structure for resources more clear like having `resources/javascript` and `resources/rust`. What do you think?

Sorry this process has taken so long! Hopefully we can get this merged shortly and Rust dev can take off.

**Comment by johnnyman727** [comment]
@nplus11 are you still working on this?

File: resources/main.rs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  
+  // Set the led pins as outputs with initial states
+  let led1 = tessel.led[0].output(1);
+  let led2 = tessel.led[1].output(0);
+
+  // Toggle the led states
+  loop {
+    println!("I'm blinking! (Press CTRL + C to stop)");
+    led1.toggle();
+    led2.toggle();
```
Comment (rwaldron) [inline resources/main.rs]: Is there latency in `loop` that would make this toggling noticeable?

**Comment by rwaldron** [comment]
This will need to be rebased again and ideally squashed

**Comment by johnnyman727** [comment]
This PR is beyond saving :( I tried rebasing and it was a mess. I'm going to manually make another PR so @nplus11 work isn't all for naught.

**Comment by johnnyman727** [comment]
Closing in favor of https://github.com/tessel/t2-cli/pull/292.

**Pull request closed without merging**